File: python/services/repository/ports/identity_client.py
"""IdentityClient port — validates user existence via the identity service (A.3)."""
from __future__ import annotations

from typing import Protocol, runtime_checkable


@runtime_checkable
class IdentityClient(Protocol):
    """Driven port for confirming a user exists before persisting a repository."""

    async def validate_user_exists(self, user_id: int) -> None:
        """Return None when the user exists and is active.

        Raise ERR_OWNER_NOT_FOUND when not found, ERR_INTERNAL on transport error.
        """
        ...

File: core/services/migration/ports/identity_client.go
package ports

import "context"

// IdentityClient is the driven port for validating user existence via the
// identity service. The migration service calls this before creating a migration
// to confirm the owner_user_id refers to an active user.
type IdentityClient interface {
	// ValidateUserExists returns nil when the user exists and is active,
	// domain.ErrOwnerNotFound when the user is not found, or domain.ErrInternal
	// for unexpected transport errors.
	ValidateUserExists(ctx context.Context, userID uint64) error
}
